### PR TITLE
fix(settings): Deprecate old things for real

### PR DESCRIPTION
**Pull request by nickvergessen**
Psalm output before:
```
ERROR: InvalidDocblock - lib/public/Settings/IManager.php:40:2 - Typo in @deprecated for const in OCP. (see https://psalm.dev/008)
	/**
	 * @since 9.1.0
	 * @depreacted 29.0.0 Use {@see self::SETTINGS_ADMIN} instead
	 */
	public const KEY_ADMIN_SETTINGS = 'admin';


ERROR: InvalidDocblock - lib/public/Settings/IManager.php:46:2 - Typo in @deprecated for const in OCP. (see https://psalm.dev/008)
	/**
	 * @since 9.1.0
	 * @depreacted 29.0.0 Use {@see self::SETTINGS_ADMIN} instead
	 */
	public const KEY_ADMIN_SECTION = 'admin-section';


ERROR: InvalidDocblock - lib/public/Settings/IManager.php:52:2 - Typo in @deprecated for const in OCP. (see https://psalm.dev/008)
	/**
	 * @since 13.0.0
	 * @depreacted 29.0.0 Use {@see self::SETTINGS_PERSONAL} instead
	 */
	public const KEY_PERSONAL_SETTINGS = 'personal';


ERROR: InvalidDocblock - lib/public/Settings/IManager.php:58:2 - Typo in @deprecated for const in OCP. (see https://psalm.dev/008)
	/**
	 * @since 13.0.0
	 * @depreacted 29.0.0 Use {@see self::SETTINGS_PERSONAL} instead
	 */
	public const KEY_PERSONAL_SECTION = 'personal-section';


------------------------------
4 errors found
------------------------------
```

## Follow up

- Also enforce the `@since` for constants in OCP => Follow up PR as there are 223 failures 😱 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by bigcat88 — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]
:elephant:

File: build/psalm/OcpSinceChecker.php [hunk 1]
```diff
@@ -111,5 +124,58 @@ private static function checkMethodComment(Stmt $stmt, FileSource $statementsSou
...
+	private static function checkConstantComment(Stmt $stmt, FileSource $statementsSource): void {
+		$docblock = $stmt->getDocComment();
+
+		if ($docblock === null) {
+//			IssueBuffer::maybeAdd(
+//				new InvalidDocblock(
+//					'PHPDoc is required for const in OCP.',
+//					new CodeLocation($statementsSource, $stmt)
+//				),
+//			);
```
Comment (blizzz) [inline build/psalm/OcpSinceChecker.php:147]: ?

File: build/psalm/OcpSinceChecker.php [hunk 2]
```diff
@@ -111,5 +124,58 @@ private static function checkMethodComment(Stmt $stmt, FileSource $statementsSou
...
+
+//		if (!isset($parsedDocblock->tags['since'])) {
+//			IssueBuffer::maybeAdd(
+//				new InvalidDocblock(
+//					'@since is required for const in OCP.',
+//					new CodeLocation($statementsSource, $stmt)
+//				)
+//			);
+//		}
+
```
Comment (blizzz) [inline build/psalm/OcpSinceChecker.php:171]: ?

**Review by blizzz — APPROVED** [review]
not sure why commented out code was committed

**Review by AndyScherzinger — APPROVED** [review]

File: build/psalm/OcpSinceChecker.php (hunk 1 above)
Comment (nickvergessen) [reply build/psalm/OcpSinceChecker.php:147]: See todo for followup PR 😅

Comment (nickvergessen) [reply build/psalm/OcpSinceChecker.php:147]: Okay, CS is failing so I will fix everything right away

**Review by st3iny — APPROVED** [review]

**Pull request merged**